x-pack/libbeat/autodiscover/providers/aws/config.go

```diff
@@ -0,0 +1,41 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package aws
+
+import (
+	"time"
+
+	"github.com/elastic/beats/x-pack/libbeat/common/aws"
+
+	"github.com/elastic/beats/libbeat/autodiscover/template"
+	"github.com/elastic/beats/libbeat/common"
+)
+
+// Config for all aws autodiscover providers.
+type Config struct {
+	Type string `config:"type"`
+
+	// Standard autodiscover fields.
+
+	// Hints are currently not supported, but may be implemented in a later release
+	HintsEnabled bool                    `config:"hints.enabled"`
+	Builders     []*common.Config        `config:"builders"`
+	Appenders    []*common.Config        `config:"appenders"`
+	Templates    template.MapperSettings `config:"templates"`
```

Comment:
You can remove these if they are not implemented, hints doesn't always make sense as sometimes there is no good mechanism to store hints in the source data. This is probably the case here

Reply:
I actually copied this part from elb provider and thinking to use it for both aws_ec2 and aws_elb provider. Let me double check if elb is using hints, if not, I will remove this. Thanks!

Reply:
These config variables are removed now.